File: ts-sdk/src/sdk/models/operations/listaccounts.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class ListAccountsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ledger" })
  ledger: string;
}
export enum ListAccountsBalanceOperatorEnum {
    Gte = "gte",
    Lte = "lte",
    Gt = "gt",
    Lt = "lt",
    E = "e",
    Ne = "ne"
}

export class ListAccountsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=address" })
  address?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=after" })
  after?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=balance" })
  balance?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=balanceOperator" })
  balanceOperator?: ListAccountsBalanceOperatorEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=deepObject;explode=true;name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pagination_token" })
  paginationToken?: string;
}

export class ListAccountsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListAccountsPathParams;

  @SpeakeasyMetadata()
  queryParams: ListAccountsQueryParams;
}

export class ListAccountsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  accountsCursorResponse?: shared.AccountsCursorResponse;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorResponse?: shared.ErrorResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/listaccounts.go
package operations

import (
	"github.com/speakeasy-sdks/formance-sdks/go-client-sdk/pkg/models/shared"
)

type ListAccountsPathParams struct {
	Ledger string `pathParam:"style=simple,explode=false,name=ledger"`
}

type ListAccountsBalanceOperatorEnum string

const (
	ListAccountsBalanceOperatorEnumGte ListAccountsBalanceOperatorEnum = "gte"
	ListAccountsBalanceOperatorEnumLte ListAccountsBalanceOperatorEnum = "lte"
	ListAccountsBalanceOperatorEnumGt  ListAccountsBalanceOperatorEnum = "gt"
	ListAccountsBalanceOperatorEnumLt  ListAccountsBalanceOperatorEnum = "lt"
	ListAccountsBalanceOperatorEnumE   ListAccountsBalanceOperatorEnum = "e"
	ListAccountsBalanceOperatorEnumNe  ListAccountsBalanceOperatorEnum = "ne"
)

type ListAccountsQueryParams struct {
	Address         *string                          `queryParam:"style=form,explode=true,name=address"`
	After           *string                          `queryParam:"style=form,explode=true,name=after"`
	Balance         *int64                           `queryParam:"style=form,explode=true,name=balance"`
	BalanceOperator *ListAccountsBalanceOperatorEnum `queryParam:"style=form,explode=true,name=balanceOperator"`
	Cursor          *string                          `queryParam:"style=form,explode=true,name=cursor"`
	Metadata        map[string]interface{}           `queryParam:"style=deepObject,explode=true,name=metadata"`
	PageSize        *int64                           `queryParam:"style=form,explode=true,name=pageSize"`
	PaginationToken *string                          `queryParam:"style=form,explode=true,name=pagination_token"`
}

type ListAccountsRequest struct {
	PathParams  ListAccountsPathParams
	QueryParams ListAccountsQueryParams
}

type ListAccountsResponse struct {
	AccountsCursorResponse *shared.AccountsCursorResponse
	ContentType            string
	ErrorResponse          *shared.ErrorResponse
	StatusCode             int64
}
